### PR TITLE
Add generic SortedItemList.

### PR DESCRIPTION
**Pull request by arknotts**
This lets consuming code just derive from the SortedItemList and items will automatically be sorted by their display property.

### CODE REVIEW

File: source/types/itemList.ts [hunk 1]
```diff
@@ -38,3 +38,11 @@ export class ItemList<TItemType extends IItem> {
 		return this.items;
 	}
 }
+
+export class SortedItemList<TItemType extends IItem> extends ItemList<TItemType> {
+	setItems(items: TItemType[]): void {
+		super.setItems(items.sort((i1, i2) => {
+			return i1.display.localeCompare(i2.display);
```
Comment (KorinaGraber) [inline source/types/itemList.ts]: What's the benefit of this over a simple `i1 === i2`?

Comment (arknotts) [inline source/types/itemList.ts]: That'd be cleaner but unfortunately sort takes a function that should return -1, 0, or 1 depending on whether `i1<i2`, `i1 === i2`, or `i1>i2` respectively. This seemed a better way than doing all the if checks manually.

Comment (KorinaGraber) [inline source/types/itemList.ts]: How about https://lodash.com/docs#sortBy? Might be simpler in the end. Something like

```
// top of file
import { sortBy } from 'lodash';

// setItems
sortBy(items, 'display');
or
sortBy(items, x => x.display);
```

Comment (arknotts) [inline source/types/itemList.ts]: Ah I do like that better. I looked at the lodash docs originally but missed the one that takes a function and didn't like using `'display'` as a string (not very maintainable). But the typescript arrow function makes it look much cleaner. New commit pushed.

Comment (KorinaGraber) [inline source/types/itemList.ts]: :+1:

**Pull request merged**